### DISCUSSION TITLE
Come funziona il permesso amministrativo di cambio password degli utenti e quando può essere attivato?

### DISCUSSION THREAD

**Discussion by marcomeneghin** [globaleaks/globaleaks-whistleblowing-software#3910] [Q&A] [answered]
Posso impostare questa funzionalità in qualsiasi momento e mi abilita a gestire la pwd anche su utenti esistenti e che hanno fatto il primo accesso?
![image](https://github.com/globaleaks/GlobaLeaks/assets/131891399/4d25707e-136d-4abd-a367-3e18c2e44c5d)
Gli utenti possono attivarsi in autonomia la chiave di recupero ma la probabilità che la perdano è elevata.
Voglio essere certo che inqualsiasi momento sono in grado di far accedere un utente che si fosse scordato le sue credenziali/chiave di recupero o impossibilitato ad usare MFA.

Grazie, M.

**Comment by evilaliv3** [comment, marked as answer]
La funzionalità può essere attivata in ogni momento ma dal momento della sua attivazione sarà attiva e proteggerà gli account dei soli utenti che effettueranno un login.

Scenario di buon funzionamento:
- La funzionalità viene attivata da un amministratore
- Gli utenti effettuano un primo login in cui vengono inviati a cambiare la password e la loro ciave viene protetta dalle chiavi di escrow generate
- Gli utenti perdono la loro password e la loro chiave di recupero
- L'amministratore riesce a ripristinare gli account

Scenario di cattivo funzionamento
- La funzionalità non viene attivata o viene attivata quando gli utenti hanno già perso la loro password e la loro chiave di recupero
- L'amministratore non ha alcun modo di ripristinare gli account

**Comment by marcomeneghin** [comment]
Nel secondo caso se l'utente che ha perso le chiavi è l'unico gestore e ci sono segnalazioni aperte come posso creare un nuovo gestore e dargli l'accesso alla segnalazione rimasta "orfana*?

In ogni caso mi sembra di capire che per essere tranquilli e poter sempre recuperare un utente devo attivare questa funzionalità

Grazie delle informazioni e del supporto.

Marco.

**Comment by evilaliv3** [comment]
Non esiste soluzione nel secondo caso,

I sistemi Il sistema di escrow come quello che state analizzando è l'unica garanzia aggiuntiva, accettabile in un contesto di servizio o a tutela di situazioni di conflitto di interesse o per resilienza a perdite umane. Permettiamo di mantenerlo eventualmente disattivato perchè in si allontana da un modello crittografico perfetto  perchè un sistema cifrato per definizione deve prevede che l'unico modo di accedere al contenuto sia avere accesso alla chiave crittografica.

**Comment by marcomeneghin** [comment]
Grazie, Chiaro,
  Noi configuriamo il sistema sempre con sito padre per l'amministratore e siti figli per la gestione delle segnalazioni delle singole aziende.
 Solo il sito ladre ha un utente amministratore. É sufficiente impostare nel padre la possibilità di impostare nuova password agli utenti e questo a cascata può farlo anche su tutti gli utenti di tipo gestore delle segnalazioni nei siti figlio anche se li l'opzione è disattivata? Corretto?

Vista l'importanza fondamentale della chiave di recupero io penserei a rendere obbligatorio visualizzarla almeno in fase di primo accesso.

Grazie ancora del supporto.

Marco.

**Comment by evilaliv3** [comment]
Grazie della domanda e dei feedback.

Confermiamo che l'abilitazione sull'istanza principale garantisce il recupero delle secondarie. Assicuratevi in questo caso di avere almeno due utenti amminsitratori abilitati al cambio password che è buona pratica.

Circa il suggerimento sulla visualizzazione della chiave di recupero c'è uno specifico warning in homepage che viene mostrato agli utenti finche non effettuano l'operazione.

Magari si potremmo aggiungere un specifico step del wizard che spiega questi aspetti.

Giovanni
